File: String/13-RomanToInteger.py
class Solution:
    def romanToInt(self, s: str) -> int:

        l = len(s)-1
        sum = 0
        Store = {"I": 1, "V": 5, "X": 10,
                 "L": 50, "C": 100, "D": 500, "M": 1000}
        for i in range(0, l):
            if(Store[s[i]] < Store[s[i+1]]):
                sum -= Store[s[i]]
            else:
                sum += Store[s[i]]
        sum = sum+Store[s[l]]
        return sum


f = Solution()
print(f.romanToInt("MCMXCIV"))

File: LeetCode/String/13-RomanToInteger.go
package main

import "fmt"

func romanToInt(s string) int {

	m := map[byte]int{'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}

	p, c, sum := 0, 0, 0

	l := len(s) - 1

	for i := 0; i < l; i++ {
		c = m[s[i]]
		if p < c {
			sum -= m[s[i]]
		} else {
			sum += m[s[i]]
		}
		p = c
	}
	return sum
}

func main() {
	fmt.Println(romanToInt("III"))
}
